### ISSUE TITLE
When MPAS throws an exception or atm tasks fail, status is set to success

### ISSUE THREAD

**Issue by sterlingbaldwin** [ACME-Climate/a-prime#11]
I just started a run (using my own conda environment) and even though MPAS threw an exception and no diagnostics were produced, both the atm and ocnice diagnostics reported success.

```
MPAS-Analysis submodule:  30b394a73fc47f5d8f25f9c1e5b5685ed529cc74 python/MPAS-Analysis (v0.5)
Traceback (most recent call last):
  File "python/MPAS-Analysis/run_mpas_analysis.py", line 13, in <module>
    import matplotlib as mpl
ImportError: No module named matplotlib
ocnice diags complete
```

Additionally when the atm diags were trying to run, turns out I put my data in the wrong directory, but when it exited it also reported it was complete

```
/p/cscratch/acme/baldwin32/aprime_test_20171107/20171011.beta2_FCT2-icedeep_branch.A_WCYCL1850S.ne30_oECv3_ICG.edison/run for 20171011.beta2_FCT2-icedeep_branch.A_WCYCL1850S.ne30_oECv3_ICG.edison does not exist! Please check.
Exiting atmosphere diagnostics...

atm diags complete
```

```
Status of atmospheric diagnostics: 0
 (0-->success, -1-->diags not invoked, -2-->batch_script, 3-->init error)
Status of ocean/ice diagnostics: 0
 (0-->success, -1-->diags not invoked, -2-->batch_script, 3-->init error)
```

If either one of these fails, the script should probably exit with an error code, or at least set their status reported to the user to something other then success.

**Comment by xylar** [comment]
I also have seen cases where scripts within the atmosphere diagnostics fail but I don't see any indication of a problem until I look in the log file.  The only other way to know something went wrong is that there is no plot in the end.

**Comment by milenaveneziani** [comment]
We have addressed as much of this issue as possible in #24. I will close this now and we can open a new issue if other instances occur, so that we can address them specifically.
